storage/src/checker/mod.rs

```diff
@@ -224,9 +225,10 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
                 let next_addr = current_addr
                     .checked_add(*area_size)
                     .expect("address overflow is impossible");
-                if next_addr <= leaked_range.end {
+                if next_addr <= *leaked_range.end {
                     leaked.push((current_addr, area_index as AreaIndex));
-                    current_addr = next_addr;
+                    current_addr =
+                        LinearAddress::new(next_addr.get()).expect("next_addr is non-zero");
```

Comment:
same here, `next_addr` should already be a `LinearAddress` by this point and we should not need to call expect.